systems-analysis/cache-task-support.md

```diff
@@ -0,0 +1,847 @@
+# CacheTask Support in Dragonfly
+
+## Overview
+
+This design document aims to extend the task processing capabilities of the Dragonfly client by introducing a new task type called CacheTask, which enables file distribution based entirely on in-memory caching. Unlike the existing Task type, CacheTask completely bypasses the disk storage layer (Content Storage) and interacts only with the in-memory cache layer (Cache Storage), thereby eliminating disk I/O bottlenecks and significantly improving the distribution performance of hot files.
+
+## Motivation
+
+- **Performance**: Pure in-memory operations eliminate disk I/O overhead, providing faster data access compared to existing cache systems that require disk backup.
+- **Efficiency**: Avoids complex disk garbage collection logic.
+- **Scalability**: Dynamic memory management supports more flexible caching strategies, allowing cache files to be replaced by algorithms such as LRU, and prevents system service disruption due to insufficient capacity.
+- **Simplicity**: The pure in-memory architecture reduces the complexity of file system operations and avoids the overhead of disk path management.
+
+## Design Goals
+
+- Add support for CacheTask while maintaining compatibility with the existing Task and PersistentCacheTask implementations.
+- Retain the current storage interfaces to ensure seamless integration with the existing task management system.
+- Support Dragonfly’s pure in-memory caching mechanism without disk persistence.
+- Add a dfget parameter to simplify the creation and management of CacheTask.
+
+## Architecture
+
+### Current Architecture
+
+```mermaid
+flowchart TD
+    dfdaemon --> Task
+    dfdaemon --> PersistentCacheTask["Persistent Cache Task"]
+    
+    Task --> SchedulerClient1["Scheduler Client"]
+    Task --> StorageSystem1["Storage System"]
+    Task --> PieceManager1["Piece Manager"]
+    
+    StorageSystem1 --> Content1["Content"]
+    StorageSystem1 --> Cache1["Cache"]
+    StorageSystem1 --> Metadata1["Metadata"]
+    
+    PersistentCacheTask --> SchedulerClient2["Scheduler Client"]
+    PersistentCacheTask --> StorageSystem2["Storage System"]
+    PersistentCacheTask --> PieceManager2["Piece Manager"]
+    
+    StorageSystem2 --> Content2["Content"]
+    StorageSystem2 --> Metadata2["Metadata"]
+```
+
+### Proposed Architecture
+
+```mermaid
+flowchart TD

```

Comment:
![image](https://github.com/user-attachments/assets/3e3b402c-d62a-4129-aa92-7cd41c88d676)

Task only calls Content (disk), not Cache (memory).